openedx/core/djangoapps/content/learning_sequences/data.py

```diff
@@ -365,3 +385,50 @@ class UserCourseOutlineDetailsData:
     outline: UserCourseOutlineData
     schedule: ScheduleData
     special_exam_attempts: SpecialExamAttemptData
+
+
+# Length, in characters, of a base64-encoded usage key hash.
+# These hashes are being experimentally used to shorten courseware URLs.
+# See TNL-8638 for more details.
+USAGE_KEY_HASH_LENGTH = 8
+
+# Number of hash digest bits needed in order to produce a base64-encoded output
+# of length `USAGE_KEY_HASH_LENGTH`.
+# Each base64 character captures 6 bits (because 2 ^ 6 = 64).
+_USAGE_KEY_HASH_BITS = USAGE_KEY_HASH_LENGTH * 6
+
+# Number of hash digest bytes needed in order to produce a base64-encoded output
+# of length `USAGE_KEY_HASH_LENGTH`. Is equal to one eighth of `_USAGE_KEY_HASH_BITS`.
+# In the event that _USAGE_KEY_HASH_BITS is not divisible by 8, we round up.
+# We will cut off the extra any output at the end of `hash_usage_key`.
+_USAGE_KEY_HASH_BYTES = math.ceil(_USAGE_KEY_HASH_BITS / 8)
+
+# A regex to capture strings that could be the output of `hash_usage_key`.
+# Captures a string of length `USAGE_KEY_HASH_LENGTH`, made up of letters,
+# numbers, dashes, underscores, and/or equals signs.
+USAGE_KEY_HASH_PATTERN = rf'(?P<usage_key_hash>[A-Z_a-z0-9=-]{{{USAGE_KEY_HASH_LENGTH}}})'
```

Comment:
Doesn't the linter complain if you declare these constants in the middle of the module? If not, that works. 🤷

Reply:
No complaints! Maybe because they come before any function definitions?